helper/restore_helper.go

```diff
@@ -329,6 +362,14 @@ func doRestoreAgent() error {
 			}
 			log(fmt.Sprintf("Oid %d: Copied %d bytes into the pipe", oid, bytesRead))
 
+			// On resize restore reader might be nil.
+			if !*singleDataFile && !(*isResizeRestore && contentToRestore >= *origSize) {
+				if err = readers[contentToRestore].waitForPlugin(); err != nil {
```

Comment:
A plugin process can do its own internal stuff after it has complete copying data, but before it has exited, and it might take some time (unpredictable in general case, as we can't predict what kind of plugins we may use in future). And in case non-single data file, we will include this time X number of tables (which could be pretty large) into the gprestore execution time.

So, my question is - should we expect gprestore execution time degradation with this change? At what magnitude? At least fot the currently used `adb_ddp_plugin`.

Reply:
Please take a look at the [discussion](https://tracker.yandex.com/ADBDEV-5531#66795307657ae13850cd5543) from the tracker.

Reply:
As discussed offline, with adb_ddp_plugin the restore doesn't seem to degrade in performance, so closing this question for now.